webhooks-with-pgstream.mdx

```diff
@@ -0,0 +1,555 @@
+
+## Configuration
+
+The `pgstream` repo comes with a `pg2webhook.env` configuration file for this purpose. Its contents are:
+
+```sh
+# Listener config
+PGSTREAM_POSTGRES_LISTENER_URL="postgres://postgres:postgres@localhost?sslmode=disable"
+
+# Processor config
+PGSTREAM_TRANSLATOR_STORE_POSTGRES_URL="postgres://postgres:postgres@localhost?sslmode=disable"
+PGSTREAM_WEBHOOK_SUBSCRIPTION_STORE_URL="postgres://postgres:postgres@localhost?sslmode=disable"
+PGSTREAM_WEBHOOK_SUBSCRIPTION_STORE_CACHE_ENABLED=false
+PGSTREAM_WEBHOOK_SUBSCRIPTION_STORE_CACHE_REFRESH_INTERVAL="60s"
+```
+
+In this file, there are two Postgres connection strings defined. The first one is for the listener, to which database should pgstream connect and listen for changes. The second one is used as a metadata db to store the registered subscriptions and their configuration. In this example, they are both set to the same database, but you can use different databases or instances if you prefer.
+
+You are now ready to start `pgstream` like this:
+
+```sh
+pgstream run -c pg2webhook.env
+```
+
+## The webhooks listener
+
+We now have pgstream ready to go, but we need something to receive the webhooks. Let's use this simple python program (courtesy of ChatGPT) to start a webserver that logs all the requests it receives:
+
+```python
+import http.server
+import json
+from urllib.parse import urlparse, parse_qs
+
+class WebhookHandler(http.server.BaseHTTPRequestHandler):
+    def do_POST(self):
+        # Parse the length of the data
+        content_length = int(self.headers['Content-Length'])
+
+        # Read the data from the request
+        post_data = self.rfile.read(content_length)
+
+        # Try to parse the data as JSON
+        try:
+            json_data = json.loads(post_data)
+            pretty_json = json.dumps(json_data, indent=4)
+            print("Received Webhook:")
+            print(pretty_json)
+        except json.JSONDecodeError:
+            print("Received non-JSON data:")
+            print(post_data.decode('utf-8'))
+
+        # Send a response back to the webhook sender
+        self.send_response(200)
+        self.send_header('Content-Type', 'text/plain')
+        self.end_headers()
+        self.wfile.write(b'Webhook received')
+
+def run(server_class=http.server.HTTPServer, handler_class=WebhookHandler, port=9910):
+    server_address = ('', port)
+    httpd = server_class(server_address, handler_class)
+    print(f'Starting server on port {port}...')
+    httpd.serve_forever()
+
+if __name__ == '__main__':
+    run()
+```
+
+You can start it like this:
+
+```sh
+python3 webhooks_server.py
+
+Starting server on port 9910...
+```
+
+Now we have to create a subscription so that `pgstream` knows to send the webhook to `localhost:9910` where the `webhook_server.go` program is waiting. We do this by calling an API which the webhook module from `pgstream` serves:
```

Comment:
Here maybe we should explain that we can create subscriptions for any table in a schema, or for all schemas in a database by leaving those fields empty. The same applies to the `event_types`, if none are provided, any event type will trigger a notification, but we can specify the event type (`I`, `U`,`D` or `T`) if we only care about some of them.

Reply:
Added a Note with these two points. Can you review that part, please?